src/test/java/com/cloudbees/jenkins/support/CheckFilterTest.java

```diff
@@ -0,0 +1,412 @@
+        // Create the objects needed for some contents to be included
+        QueueTaskFuture<FreeStyleBuild> build = createObjectsWithNames();
+
+        // Reload the mappings, after the objects were created
+        ContentFilters.get().setEnabled(true);
+        ContentFilter.ALL.reload();
+        // Generate the filtered words
+        checker.generateFilteredWords();
+
+        // Check the components are filtered correctly
+        assertComponent(Arrays.asList(  AboutJenkins.class,
+                                        AboutUser.class,
+                                        AgentsConfigFile.class,
+                                        BuildQueue.class,
+                                        ConfigFileComponent.class,
+                                        DumpExportTable.class,
+                                        EnvironmentVariables.class,
+                                        JVMProcessSystemMetricsContents.Agents.class,
+                                        JVMProcessSystemMetricsContents.Master.class,
+                                        SystemConfiguration.class,
+                                        NetworkInterfaces.class,
+                                        NodeMonitors.class,
+                                        UpdateCenter.class,
+                                        SystemProperties.class,
+                                        ThreadDumps.class
+                                    ), checker);
+
+        // Cancel the job running
+        build.cancel(true);
+    }
+
+    private QueueTaskFuture<FreeStyleBuild> createObjectsWithNames() throws Exception {
+        // For an environment variable
+        User.getOrCreateByIdOrFullName("path");
+
+        // For JVMProcessSytemMetricsContents
+        User.getOrCreateByIdOrFullName("kb");
+        User.getOrCreateByIdOrFullName("mb");
+        User.getOrCreateByIdOrFullName("max");
+        User.getOrCreateByIdOrFullName("mounted");
+
+        // ThreadDumps
+        User.getOrCreateByIdOrFullName("runnable");
+
+        // For SystemProperties
+        User.getOrCreateByIdOrFullName("encoding");
+
+        // For ConfigFileComponent --> config.xml
+        jenkins.jenkins.getView("all").rename(VIEW_ALL_NEW_NAME);
+        jenkins.jenkins.save();
+
+        // Create the slave slave0 for some components
+        jenkins.createOnlineSlave();
+
+        // Create a job to have something pending in the queue for the BuilQueue component
+        FreeStyleProject project = jenkins.createFreeStyleProject(JOB_NAME);
+        project.setAssignedLabel(new LabelAtom("foo")); //it's mandatory
+
+        //project.getBuildersList().add(new SimpleBuilder());
+        QueueTaskFuture<FreeStyleBuild> buildQueued = project.scheduleBuild2(0);
+        return buildQueued;
+    }
+
+    private void assertComponent(List<Class<? extends Component>> componentClasses, FileChecker checker) throws IOException {
```

Comment:
Create a bundle with the components specified and check with the ```checker``` object the contents of the files in the bundle